### ISSUE TITLE
Floating point exception (core dumped)

### ISSUE THREAD

**Issue by zoushipeng** [microsoft/VibeVoice#71]
when I run this:
python demo/inference_from_file.py --model_path microsoft/VibeVoice-1.5B --txt_path demo/text_examples/1p_abs.txt --speaker_names Alice

I get some error, I find it occurs at 
VibeVoiceForConditionalGenerationInference/forward/logits = self.lm_head(hidden_states[:, slice_indices, :])

I print "hidden_states[:, slice_indices, :]"  then I get 
tensor([[[ 0.4473, -1.0312,  0.4863,  ...,  0.8867,  0.9414, -2.4688]]],
       device='cuda:0', dtype=torch.bfloat16)

so what's wrong ?

**Comment by YaoyaoChang** [comment]
Env? Error message?

**Comment by YaoyaoChang** [comment]
Looks like https://github.com/microsoft/VibeVoice/pull/72 fixed your problem — try it out and share how it goes!

**Comment by zoushipeng** [comment]
thanks very much. I will try it later~

**Comment by vatsalm1611** [comment]
issue: 71
audience: reporter_and_maintainers
note: >
  Quick tester note — I opened PR #72 with two minimal guards at the inference
  boundary to address the CUDA bf16 crash:
  (1) validate slice_indices before slicing (clear Python error instead of a
  CUDA FPE), and (2) if running on CUDA with bfloat16 on a GPU without native
  bf16 (SM < 80), upcast only at the lm_head call.

guards:
  - "Validate `slice_indices` before slicing (raise clear Python error)."
  - "On CUDA with bfloat16 + SM < 80: upcast only at `lm_head` boundary."

try_branch:
  branch: "vatsalm1611:fix/cuda-bf16-fpe-guard"

commands:
  fetch_and_checkout: |
    git fetch https://github.com/vatsalm1611/VibeVoice.git \
      fix/cuda-bf16-fpe-guard:tmp-fpe-guard
    git checkout tmp-fpe-guard
  repro: |
    python demo/inference_from_file.py \
      --model_path microsoft/VibeVoice-1.5B \
      --txt_path demo/text_examples/1p_abs.txt \
      --speaker_names Alice

if_still_fails:
  ask_for_env: |
    python - <<'PY'
    import torch, platform
    print("torch:", torch.__version__)
    print("cuda:", torch.version.cuda)
    print("device:", torch.cuda.get_device_name(0))
    print("sm:", torch.cuda.get_device_capability(0))
    print("bf16_supported:", torch.cuda.is_bf16_supported())
    print("python:", platform.python_version())
    PY
  also_include:
    - "OS name and version"
    - "NVIDIA driver version"
    - "Exact error text / stacktrace"

closing: >
  Thanks! These details will help us compare environments and confirm whether
  PR #72 fully addresses the crash on your setup.

**Comment by vatsalm1611** [comment]
**Quick tester note** (for @zoushipeng and anyone who can repro)

**Context.** Issue #71 hits a “Floating point exception (core dumped)” on CUDA with `torch.bfloat16`, crashing at the `lm_head` call on sliced hidden states.

**PR #72** adds two minimal guards:
1. Validate `slice_indices` before slicing (so invalid/empty selections raise a clear Python error instead of a CUDA kernel FPE).
2. If **CUDA + bf16** on a GPU *without* native bf16 (SM < 80), upcast to `float32` **only** at the `lm_head` boundary.

---

### Try the fix
```bash
git fetch https://github.com/vatsalm1611/VibeVoice.git fix/cuda-bf16-fpe-guard:tmp-fpe-guard
git checkout tmp-fpe-guard

python demo/inference_from_file.py \
  --model_path microsoft/VibeVoice-1.5B \
  --txt_path demo/text_examples/1p_abs.txt \
  --speaker_names Alice

**Comment by zoushipeng** [comment]
it not worked,see   https://github.com/microsoft/VibeVoice/pull/72#issuecomment-3242375852
